### PR TITLE
feat: apply the wasm module of lbm-sdk

### PR DESCRIPTION
**Pull request by zemyblue**
Apply all the wasm modules changes of lbm-sdk.
The `x/wasm` module is copied from the [`3bdcb6ffe01c81615bedb777ca0e039cc46ef00c`](https://github.com/line/lbm-sdk/tree/3bdcb6ffe01c81615bedb777ca0e039cc46ef00c) commit hash in lbm-sdk. And some parts is modified to fit properly. The modified part can be seen in the differences below.

## The difference with lbm-sdk/x/wasm
* use wamd's app instead of simapp of lbm-sdk
* skip ibc unittest
* add `flagProposalType` and `cli.FlagProposal` in gov tx cli (`x/wasm/client/cli/gov_tx.go`). This flags was added in cosmwasm/wasmd.
  * `ProposalStoreCodeCmd()`
  * `ProposalInstantiateContractCmd()`
  * `ProposalMigrateContractCmd()`
  * `ProposalExecuteContractCmd()`
  * `ProposalSudoContractCmd()`
  * `ProposalUpdateContractAdminCmd()`
  * `ProposalClearContractAdminCmd()`
  * `ProposalPinCodesCmd()`
  * `ProposalUnpinCodesCmd()`
  * `ProposalUpdateInstantiateConfigCmd()`
* copy `ReadFileWithSizeLimit` function of lbm-sdk's internal file under `x/wasm/client/cli/os` directory. This is because functions inside `internal` directory cannot be imported.

## Update process
1. import zemyblue/lbm-sdk in line/wasmd (in this PR)
2. register the PR which remove `x/wasm` in lbm-sdk. This lbm-sdk's PR imports current PR temporary and replace `line/wasmd` after merging current PR. 
3. register a PR to replace the import path of `line/wasmd` from `zemyblue/lbm-sdk` to `line/lbm-sdk` (via another PR in this repo)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,28 +24,28 @@ A VM can support one or more contract-VM interface versions. The interface
...
-| 0.16  |              | 0.14        | 0.14         |
-| 0.15  |              | 0.13        | 0.11-0.13    |
-| 0.14  |              | 0.13        | 0.11-0.13    |
-| 0.13  |              | 0.12        | 0.11-0.13    |
-| 0.12  |              | 0.12        | 0.11-0.13    |
-| 0.11  |              | 0.11        | 0.11-0.13    |
-| 0.10  |              | 0.10        | 0.10         |
-| 0.9   |              | 0.9         | 0.9          |
-| 0.8   |              | 0.8         | 0.8          |
+| line/wasmd | cosmwasm/wasmd | line/wasmvm   | cosmwasm/wasmvm | cosmwasm-vm | cosmwasm-std |
```
Comment (loloicci) [inline README.md:27]: Is this `cosmwasm/wasmvm` needed?

File: README.md [hunk 2]
```diff
@@ -1,19 +1,19 @@
 # Wasm Zone
```
Comment (shiki-tak) [inline README.md:1]: Do we need to keep this README intact?
For example, cosmwasm simply attaches a link to the original repository as follows.
https://github.com/line/cosmwasm/blob/main/README.md

File: README.md (hunk 1 above)
Comment (zemyblue) [reply README.md:27]: I wrote it for checking the difference with ours.

File: README.md (hunk 2 above)
Comment (shiki-tak) [reply README.md:1]: OK, I see.

File: cmd/wasmd/root.go [hunk 3]
```diff
@@ -189,7 +191,9 @@ func (ac appCreator) newApp(
 	var cache sdk.MultiStorePersistentCache
 
 	if cast.ToBool(appOpts.Get(server.FlagInterBlockCache)) {
-		cache = store.NewCommitKVStoreCacheManager()
+		ibCacheMetricsProvider := baseapp.MetricsProvider(cast.ToBool(viper.GetBool(server.FlagPrometheus)))
```
Comment (shiki-tak) [inline cmd/wasmd/root.go:194]: In simapp, the ibCacheMetricsProvider appears to be outside of the if statement. What is the reason for this difference?

https://github.com/line/lbm-sdk/blob/v0.46.0/simapp/simd/cmd/root.go#L243

Comment (zemyblue) [reply cmd/wasmd/root.go:194]: It seems I made a mistake. 
I'll fix it. Thank you.

File: third_party/proto/ostracon/types/evidence.proto [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  ostracon.types.LightBlock         conflicting_block    = 1;
+  int64                             common_height        = 2;
+  repeated ostracon.types.Validator byzantine_validators = 3;
+  int64                             total_voting_power   = 4; // total voting weight
+  google.protobuf.Timestamp         timestamp            = 5 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+}
+
+message EvidenceList {
+  repeated Evidence evidence = 1 [(gogoproto.nullable) = false];
+}
```
[the remark this answers is not in the document]
Comment (zemyblue) [reply third_party/proto/ostracon/types/evidence.proto]: I added.

**Review by shiki-tak — APPROVED** [review]
LGTM

**Review by loloicci — APPROVED** [review]

**Review by da1suk8 — APPROVED** [review]

**Pull request merged**